meson.build

```diff
@@ -457,6 +460,9 @@ conf_data.set('PICOLIBC_TLS', thread_local_storage, description: 'use thread loc
 conf_data.set('POSIX_IO', posix_io, description: 'Use open/close/read/write in tinystdio')
 conf_data.set('POSIX_CONSOLE', posix_console, description: 'Use POSIX I/O for stdin, stdout and stderr')
 conf_data.set('ATOMIC_UNGETC', atomic_ungetc, description: 'Use atomics for fgetc/ungetc for re-entrancy')
+if not have_builtin_expect
```

Comment:
Hrm. I think, stylistically, the macro definition should probably go in newlib/libc/include/sys/cdefs.h and that picolibc.h should just get a `'#define HAVE_BUILTIN_EXPECT, so the meson.build file would just say 

```conf_data.set('HAVE_BUILTIN_EXPECT', have_builtin_expect, 'Compiler has __builtin_expect')```

cdefs.h could then do

```
#ifndef HAVE_BUILTIN_EXPECT
#define __builtin_expect(cond, exp) (cond)
#endif
```

Reply:
done in 4a02c855a0a49f64a84412a39e88c41e7845c8ae